assembly/src/parsers/ast/mod.rs

```diff
@@ -66,6 +69,53 @@ impl ProgramAst {
 
         Ok(ProgramAst { local_procs, body })
     }
+
+    /// Compiles the AST into a program that can be executed on Miden VM.
+    #[allow(unreachable_code, unused_variables)]
+    pub fn compile(
+        self,
+        context: &AssemblyContext,
+        cb_table: &mut CodeBlockTable,
+    ) -> Result<CodeBlock, AssemblyError> {
+        let Self {
+            local_procs,
+            mut body,
+        } = self;
+
+        // split the local procs into sets that should either be inlind or exported to the cb table
+        let (mut exported, inlined): (Vec<ProcedureAst>, Vec<ProcedureAst>) =
+            local_procs.into_iter().partition(|proc| proc.is_export);
```

Comment:
Programs have only internal procedures (no exported procedures). But modules do have internal and exported procedures.

Reply:
N/A, impl moved